modules/gapi/src/backends/cpu/serialization.cpp

```diff
@@ -0,0 +1,876 @@
+        ofs_ops.write((char*)op.opaque_cvpoints.data(), cvpoints_size * sizeof(cv::Point));
+//        for(uint j = 0; j < cvpoints_size; j++)
+//        {
+//            std::cout << op.opaque_cvpoints[j].x << " " << op.opaque_cvpoints[j].y << std::endl;
+//        }
+
+        //std::vector<cv::Mat> opaque_cvmats;
+        uint cvmats_size = op.opaque_cvmats.size();
+//        std::cout << "dumpGSerializedOps cvmats_size " << cvmats_size <<  std::endl;
+        ofs_ops.write((char*)&cvmats_size, sizeof(uint));
+        for(uint j = 0; j < cvmats_size; j++)
+        {
+            std::cout << op.opaque_cvmats[j].cols << " "
+                      << op.opaque_cvmats[j].rows << " "
+                      << op.opaque_cvmats[j].type()  << " "
+                      << op.opaque_cvmats[j].step[0] << " " << std::endl;
+            size_t matSizeInBytes = op.opaque_cvmats[j].rows * op.opaque_cvmats[j].step[0];
+            int mat_type = op.opaque_cvmats[j].type();
+//            std::cout << "matSizeInBytes " << matSizeInBytes <<  std::endl;
+            ofs_ops.write((char*)&op.opaque_cvmats[j].cols, sizeof(int));
+            ofs_ops.write((char*)&op.opaque_cvmats[j].rows, sizeof(int));
+            ofs_ops.write((char*)&mat_type, sizeof(int));
+            ofs_ops.write((char*)&op.opaque_cvmats[j].step[0], sizeof(int));
+            if(matSizeInBytes!=0)
+            {
+//                std::cout << "dumpGSerializedOps write mat data " << matSizeInBytes <<  std::endl;
+                ofs_ops.write((char*)op.opaque_cvmats[j].data, matSizeInBytes);
+                for(int k = 0; k < op.opaque_cvmats[j].rows * op.opaque_cvmats[j].cols; k++)
+                {
+                    //std::cout << (int)op.opaque_cvmats[j].data[k] <<  std::endl;
+                }
+            }
+        }
+
+        //std::vector<cv::Rect> opaque_cvrects;
+        uint cvrects_size = op.opaque_cvrects.size();
+//        std::cout << "dumpGSerializedOps cvrects_size " << cvrects_size <<  std::endl;
+        ofs_ops.write((char*)&cvrects_size, sizeof(uint));
+        ofs_ops.write((char*)op.opaque_cvrects.data(), cvrects_size * sizeof(cv::Rect));
+//        for(uint j = 0; j < cvrects_size; j++)
+//        {
+//            std::cout << op.opaque_cvrects[j].x << " "
+//                      << op.opaque_cvrects[j].y << " "
+//                      << op.opaque_cvrects[j].width << " "
+//                      << op.opaque_cvrects[j].height << std::endl;
+//        }
+
+    }
+}
+
+void readGSerializedOps(GSerialized &s, std::ifstream &ifs_ops)
+{
+    uint ops_size;
+    ifs_ops.read((char*)&ops_size, sizeof(uint));
+//    std::cout << "readGSerializedOps ops size " << ops_size <<  std::endl;
+    for (uint i = 0; i < ops_size;  i++)
+    {
+        //Kernel k
+        uint k_name_size;
+        ifs_ops.read((char*)&k_name_size, sizeof(uint));
+//        std::cout << "readGSerializedOps k_name_size " << k_name_size <<  std::endl;
+        std::string k_name;
```

Comment:
You can use constructor `k_name(k_name_size)` and remove `resize` (also applicable in several other places)